### Regra dos Trapézios simples

In [1]:
import numpy as np

# Função a ser integrada
def f(x):
    y= np.exp(-x)
    return y

# T(f) = I1(f) = ( f(a) + f(b) )( b - a )/2
#Extrema direita
b=10.0
#Extrema esquerda
a=0.0
#Estimativa da integral
T_1 = ( ( f(a) + f(b) )*( b - a ) )/2;

print( "O valor da integral estimada é :\nT_simples(f) = %.6f" %(T_1) )

O valor da integral estimada é :
T_simples(f) = 5.000227


Assim, o valor obtido para a integral estimada pela regra simples foi $T_{1}(e^{-x})=5.000227$&nbsp; no intervalo $[0,10]$&nbsp;.



### Regra dos Trapézios composta

In [ ]:
import numpy as np
#definindo a função
def f(x):
    y=np.exp(-x)
    return y
#a é a extrema esquerda e b é a extrema direita
def T(N,a,b):
# Comprimento de um sub-intervalo
    h=(b-a)/N
    X=np.zeros(N+1)
    Y=np.zeros(N+1)
    for i in np.arange(0,N+1,1):
        X[i]=a+h*i
    Y=f(X)
    retorno = h*( -((f(a)+f(b))/2) + np.sum(Y) )
    return retorno

Assim, o valor obtido para a integral estimada pela regra composta foi $T_{100}(x^4)= 6.4011$ no intervalo  $[0,2]$.

In [26]:
# -f faz uma ação forçada sem o prompt preguntar SIM ou NÃO
%reset -f
# DELETANDO TODAS AS VARIÁVEIS